uipyqt.py

```diff
@@ -0,0 +1,288 @@
+        self.sc.draw_hand_if_necessary(0)
+        my_hand=self.sc.hands[0]
+        self.clear_form()
+
+        self.draw_opponent_hand()
+        self.draw_table(False)
+        self.draw_my_hand(True)
+
+        self.enable_button_for_my_move()
+        self.enable_actions()
+
+    # display the form for opponent move
+    def opponent_move(self):
+        take=self.sc.play_hand(1)
+        card_from_hand=take[0]
+        cards_from_table=take[1]
+
+        self.clear_form()
+
+        self.draw_opponent_hand(card_from_hand)
+        self.draw_table(True, cards_from_table)
+        self.draw_my_hand(False)
+
+        if len(cards_from_table)>0:
+            self.last_claimed_hand = 1
+
+        self.enable_button_for_opponent_move()
+
+
+        #draw both hands if are empty
+        self.sc.draw_hand_if_necessary(0)
+        no_more_cards = self.sc.draw_hand_if_necessary(1)
+        if no_more_cards:
+            self.finish_game()
+
+
+    # uncheck any buttons other than the one clicked
+    def disable_all_but_this(self):
+        for btn in self.my_hand_buttons:
+            if not btn is self.sender():
+                btn.setChecked(False)
+
+        self.enable_actions()
+
+    # enable actions: the sum of checked button cards on the table must be equal to
+    # checked card from hand for claiming cards.
+    # No cards on the table and one hand card can be checked for laying card in hand
+    def enable_actions(self):
+        table = []
+        table_card_set = False
+        for table_btn in self.table_buttons:
+            if table_btn.isChecked():
+                table.append(table_btn.text())
+                table_card_set=True
+
+        sum_of_table=tactics.sum_of_cards(table)
+
+        hand_card_set=False
+        hand_card_value=-1
+        for hand_btn in self.my_hand_buttons:
+            if hand_btn.isChecked():
+                hand_card_set=True
+                hand_card_value=int(hand_btn.text()[:2])
+
+        self.lay_card_button.setEnabled(hand_card_set and not table_card_set)
+
+        self.claim_cards_button.setEnabled(hand_card_set and sum_of_table == hand_card_value)
+
+    # again human move
+    def pressed_OK(self):
+        self.my_move()
+
+    # remove card from hand and add it to the table
+    def pressed_lay_card(self):
+        for button in self.my_hand_buttons:
+            if button.isChecked():
+                laid_card = button.text()
+                self.sc.hands[0].remove(laid_card)
+                self.sc.table.append(laid_card)
+                break
+
+        # now it's computer's move
+        self.opponent_move()
+
+    # remove card from hand, cards from the table - and add them to pile
+    def pressed_claim_cards(self):
+        for button2 in self.my_hand_buttons:
+            if button2.isChecked():
+                card_from_hand=button2.text()
+                self.sc.hands[0].remove(card_from_hand)
+                self.sc.piles[0].append(card_from_hand)

```

Comment:
Inappropriate Intimacy - UI class mutating Scopa instance internals; There should be a separate method exposed by the Scopa class to handle that change.